src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -484,6 +484,24 @@ public DependencyGraphSpec WithReplacedSpec(PackageSpec project)
             return newSpec;
         }
 
+        public DependencyGraphSpec WithPackageSpecs(IEnumerable<PackageSpec> packageSpecs)
```

Comment:
This one is needed because down the line we still need updated PackageSpec [down the line](https://github.com/NuGet/NuGet.Client/blob/c090005d05f35725457091e253fc159067fcd300/src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs#L83) when you call RestoreRunner.GetRequests.
So it throws below error when I remove this step.

![image](https://user-images.githubusercontent.com/8766776/91651643-002f5880-ea44-11ea-906b-caa89f6c4cee.png)
![image](https://user-images.githubusercontent.com/8766776/91651648-1210fb80-ea44-11ea-99ee-fd80f3639bca.png)